ga4gh/datamodel/reads.py

```diff
@@ -75,3 +89,59 @@ def toProtocolElement(self):
         readGroup.referenceSetId = None
         readGroup.sampleId = None
         return readGroup
+
+    def getReadAlignments(self, referenceName, referenceId, start, end):
+        """
+        Returns an iterator over the specified reads
+        """
+        if referenceName is not None and referenceId is not None:
+            raise exceptions.BadReadsSearchRequestBothRefs()
+        if referenceId is not None:
+            referenceName = self._samFile.getrname(referenceId)
+        try:
+            readAlignments = self._samFile.fetch(referenceName, start, end)
+        except ValueError as exception:
+            # TODO better error and test various error conditions
+            raise exception
+        for readAlignment in readAlignments:
+            yield self.convertReadAlignment(readAlignment)
+
+    def convertReadAlignment(self, read):
+        """
+        Convert a pysam ReadAlignment to a GAReadAlignment
+        """
+        # TODO fill out remaining fields
+        # TODO refine in tandem with code in converters module
+        ret = protocol.GAReadAlignment()
+        ret.alignedQuality = read.query_qualities
+        ret.alignedSequence = read.query_sequence
+        ret.alignment = protocol.GALinearAlignment()
+        ret.alignment.mappintQuality = read.mapping_quality
+        ret.alignment.position = protocol.GAPosition()
+        ret.alignment.position.referenceName = self._samFile.getrname(
+            read.reference_id)
+        ret.alignment.position.position = read.reference_start
+        ret.alignment.cigar = []
+        for operation, length in read.cigar:
+            gaCigarUnit = protocol.GACigarUnit()
+            gaCigarUnit.operation = self._reverseCigarMap[operation]
+            gaCigarUnit.operationLength = length
+            ret.alignment.cigar.append(gaCigarUnit)
+        ret.duplicateFragment = False
```

Comment:
It's basically any field that is None or False (or otherwise has a default value from the protocol object).  Should I just not list those fields here?